src/objects.js

```diff
@@ -1,41 +1,66 @@
 const createPerson = (name, age) => {
-  // your code here
+  return {
+    name: name, 
+    age: age
+  }
 };
 
 const getName = object => {
-  // your code here
+  return object.name
 };
 
 const getProperty = (property, object) => {
-  // your code here
+  return object[property]
+  // use square bracket notation to access a property using its variable name 
 };
 
 const hasProperty = (property, object) => {
-  // your code here
+  if (object[property]){
+    return true 
+  }
+  return false
 };
 
 const isOver65 = person => {
-  // your code here
+  if (person.age > 65){ // use dot notation to retireve each key value 
+    return true
+  }
+  return false 
 };
 
 const getAges = people => {
-  // your code here
+  const ages = [];
+  for (i = 0; i < people.length; i++) { 
+    if (people[i].age) { 
+      ages.push(people[i].age);
+    }
+  }
+  return ages;
 };
 
 const findByName = (name, people) => {
-  // your code here
-};
+  return people.find(person => person.name === name)
+}
 
 const findHondas = cars => {
-  // your code here
+ return cars.filter(car => car.manufacturer === 'Honda')
 };
 
 const averageAge = people => {
-  // your code here
+  const totalAge = people.reduce((acc, person) => acc + person.age, 0)
+  const averageAge = totalAge / people.length
+  return averageAge
 };
 
-const createTalkingPerson = (name, age) => {
-  // your code here
+const createTalkingPerson = (name, age,) => {
+  const person = {
+    name,
+    age,
+    introduce: function introduce(name){
+      return `Hi ${name}, my name is ${this.name} and I am ${this.age}!`;
```

Comment:
This is a little confusing here as you are using name to be passed into the introduce function and name is already being used to declare the person's name in the object. 

You should chose a different name for the variable you pass into the introduce function for less confusion.
for example:

`````
introduce: function introduce(otherPersonsName){
      return `Hi ${otherPersonsName}, my name is ${this.name} and I am ${this.age}!`;
``````